httpgrpc/httpgrpc.go

```diff
@@ -28,6 +28,14 @@ func Errorf(code int, tmpl string, args ...interface{}) error {
 	})
 }
 
+func ErrorfWithHeaders(code int, headers []*Header, tmpl string, args ...interface{}) error {
```

Comment:
Where would this method be called from?
I am currently getting rid of all `httpgrpc.Errorf()` calls in ingester and distributor. Would like to understand where this new method would be needed.

Reply:
after our conversation, I think this PR is no longer needed. Thanks for the headsup